ravenml/data/interfaces.py

```diff
@@ -23,6 +34,98 @@ class CreateInput(object):
     for getting additional information.
 
     """
+    def __init__(self, config:dict=None, plugin_name:str=None):
+
+        if config is None or plugin_name is None:
+            raise click.exceptions.UsageError(('You must provide the --config option '
+                'on `ravenml create` when using this plugin command.'))
+        
+        self.config = config
+        
+        ## Set up Local Cache
+        # TODO: maybe create the subdir here?
+        # currently the cache_name subdir is only created IF the plugin places files there
+        self.plugin_cache = RMLCache(plugin_name)
+        
+        ## Set up Artifact Path
+        dp = config.get('dataset_path')
+        if dp is None:
+            self.plugin_cache.ensure_clean_subpath('temp')
+            self.plugin_cache.ensure_subpath_exists('temp')
+            self.dataset_path = Path(self.plugin_cache.path / 'temp')
+        else:
+            dp = Path(os.path.expanduser(dp))
+            # check if local path contains data
+            if os.path.exists(dp) and os.path.isdir(dp) and len(os.listdir(dp)) > 0:
+                if config.get('overwrite_local') or user_confirms('Local artifact storage location contains old data. Overwrite?'):
+                    shutil.rmtree(dp)
+                else:
+                    click.echo(Fore.RED + 'Dataset creation cancelled.')
+                    click.get_current_context().exit() 
+            # create directory, need exist_ok since we only delete
+            # if directory contains files
+            # TODO: protect against paths to actual files
+            os.makedirs(dp, exist_ok=True)
+            self.dataset_path = dp
+        
+        ## Set up Imageset
+        # prompt for dataset if not provided
+        imageset_list = config.get('imageset')
+        if imageset_list is None:
+            imageset_options = cli_spinner('No imageset provided. Finding imagesets on S3...', get_imageset_names)
+            imageset_list = user_selects('Choose imageset:', imageset_options, selection_type="checkbox")
+        else: 
+            for imageset in imageset_list:
+                if imageset not in get_imageset_names():
```

Comment:
call `get_imageset_names` just once and save the list to avoid hitting S3 a bunch of unnecessary times. this will save a (small) bit of money in the long run, but is also just more efficient